File: C++/1518.cpp
// LeetCode 1518. Water Bottles
// Given numBottles full water bottles, you can exchange numExchange empty bottles for one full bottle.
// Return the maximum number of water bottles you can drink.

class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int total = numBottles;  // total bottles you can drink
        while (numBottles >= numExchange) {
            int exchanged = numBottles / numExchange;  // how many new bottles you get
            total += exchanged;
            numBottles = exchanged + (numBottles % numExchange); // new full + leftover empty
        }
        return total;
    }
};

File: Python/1518.py
# Water Bottles 

class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        k = 0
        z = numBottles
        while numBottles >= numExchange:
            a = numBottles // numExchange
            k = k + a
            numBottles = a + (numBottles % numExchange)
        z = z + k 
        return z
    
print(Solution().numWaterBottles(15, 4))